From 45b0edd5900e3adf4e01a762895490784a1916dd Mon Sep 17 00:00:00 2001
From: jesusVMayor <jesus@pexego.es>
Date: Wed, 11 Feb 2015 11:26:02 +0100
Subject: [PATCH] Error en busqueda de productos.

---
 addons/stock/product.py |   34 ++++++++++++++++++++++++----------
 1 file changed, 24 insertions(+), 10 deletions(-)

diff --git a/addons/stock/product.py b/addons/stock/product.py
index 5677078..d93b70f 100644
--- a/addons/stock/product.py
+++ b/addons/stock/product.py
@@ -27,7 +27,7 @@ from openerp.tools.float_utils import float_round
 
 class product_product(osv.osv):
     _inherit = "product.product"
-        
+
     def _stock_move_count(self, cr, uid, ids, field_name, arg, context=None):
         res = dict([(id, {'reception_count': 0, 'delivery_count': 0}) for id in ids])
         move_pool=self.pool.get('stock.move')
@@ -76,7 +76,8 @@ class product_product(osv.osv):
             if type(context['location']) == type(1):
                 location_ids = [context['location']]
             elif type(context['location']) in (type(''), type(u'')):
-                domain = [('complete_name','ilike',context['location'])]
+                location = context['location'].replace('/', ' / ')
+                domain = [('complete_name', 'ilike', location)]
                 if context.get('force_company', False):
                     domain += [('company_id', '=', context['force_company'])]
                 location_ids = location_obj.search(cr, uid, domain, context=context)
@@ -84,7 +85,15 @@ class product_product(osv.osv):
                 location_ids = context['location']
         else:
             if context.get('warehouse', False):
-                wids = [context['warehouse']]
+                if type(context['warehouse']) == type(1):
+                    wids = [context['warehouse']]
+                elif type(context['warehouse']) in (type(''), type(u'')):
+                    domain = [('name', 'ilike', context['warehouse'])]
+                    if context.get('force_company', False):
+                        domain += [('company_id', '=', context['force_company'])]
+                    wids = warehouse_obj.search(cr, uid, domain, context=context)
+                else:
+                    wids = context['warehouse']
             else:
                 wids = warehouse_obj.search(cr, uid, [], context=context)
 
@@ -244,7 +253,12 @@ class product_product(osv.osv):
         if context is None:
             context = {}
         if ('location' in context) and context['location']:
-            location_info = self.pool.get('stock.location').browse(cr, uid, context['location'])
+            if type(context['location']) in (type(''), type(u'')):
+                location = context['location'].replace('/', ' / ')
+                location_id = self.pool.get('stock.location').search(cr, uid, [('complete_name', 'ilike', location)])
+            else:
+                location_id = context['location']
+            location_info = self.pool.get('stock.location').browse(cr, uid, location_id)
             fields=res.get('fields',{})
             if fields:
                 if location_info.usage == 'supplier':
-- 
1.7.9.5

