From 0e013563e6babfe046581aec581193d541c59006 Mon Sep 17 00:00:00 2001
From: jesusVMayor <jesus@pexego.es>
Date: Thu, 5 Mar 2015 19:45:10 +0100
Subject: [PATCH] [ADD]Modificada creacion de envio para evitar duplicidad de
 procurement.

---
 addons/sale/sale.py |   25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/addons/sale/sale.py b/addons/sale/sale.py
index 17712a9..3a413e3 100644
--- a/addons/sale/sale.py
+++ b/addons/sale/sale.py
@@ -711,8 +711,9 @@ class sale_order(osv.osv):
                     procurement_obj.check(cr, uid, [x.id for x in line.procurement_ids if x.state not in ['cancel', 'done']])
                     line.refresh()
                     #run again procurement that are in exception in order to trigger another move
-                    proc_ids += [x.id for x in line.procurement_ids if x.state in ('exception', 'cancel')]
-                    procurement_obj.reset_to_confirmed(cr, uid, proc_ids, context=context)
+                    line_procs = [x.id for x in line.procurement_ids if x.state in ('exception', 'cancel')]
+                    procurement_obj.reset_to_confirmed(cr, uid, line_procs, context=context)
+                    proc_ids += line_procs
                 elif sale_line_obj.need_procurement(cr, uid, [line.id], context=context):
                     if (line.state == 'done') or not line.product_id:
                         continue
-- 
1.7.9.5

